cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/type/DatasetTypeManager.java

```diff
@@ -372,7 +373,7 @@ public Boolean call() throws DatasetModuleConflictException, IOException {
               impersonator.doAs(datasetModuleId.getParent(), new Callable<Location>() {
                 @Override
                 public Location call() throws Exception {
-                  return locationFactory.create(module.getJarLocation());
+                  return Locations.getLocationFromAbsolutePath(locationFactory, module.getJarLocationPath());
```

Comment:
Adding dependency on module 'cdap-common' in cdap-proto introduces circular dependency between modules cdap-security-spi and cdap-proto.